internal/kafka/notification_msg.go

```diff
@@ -0,0 +1,76 @@
+package kafka
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/google/uuid"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+const (
+	application                  = "image-builder"
+	bundle                       = "rhel"
+	notificationMessageVersion   = "v1.2.0"
+	NotificationSuccessEventType = "launch-success"
+	NotificationFailureEventType = "launch-failed"
+)
+
+type NotificationEvent struct {
+	Payload interface{} `json:"payload"`
```

Comment:
Using empty interface is almost always a red flag in Go. We should stick with types as hard as possible.

In this case, use [RawMessage](https://pkg.go.dev/encoding/json#RawMessage) type instead. You can implement MarshalJSON method for NotificationEvent and NotificationError/NotificationSuccess types and use it efficiently while taking advantage of types.

Btw these types should go into a new package named `notifications` or `ntf` maybe, then you can drop all the Notification prefix/suffis. I mentioned that above that there should be also an interface in the newly created package that hides the logic.

Here is a working example on how to work with RawMessage: https://go.dev/play/p/LU-w97uHF-5